hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java

```diff
@@ -148,6 +154,17 @@ private void registerInvocation(ClientInvocation clientInvocation) {
         }
     }
 
+    private void checkMaxAllowedReached() {
+        if (maxAllowedConcurrentInvocations == Integer.MAX_VALUE) {
+            return;
+        }
+        long currentConcurrentInvocations = callIdIncrementer.get() - tailCallId.get();
+        if (maxAllowedConcurrentInvocations <= currentConcurrentInvocations) {
+            throw new HazelcastOverloadException("Max allowed concurrent invocations is reached : "
```

Comment:
For example, max allowed concurrent invocation count is `10`, and you submit one more. In this case, `callIdIncrementer` is `11`, you throw `HazelcastOverloadException` but `tailCallId` is `0`. Then the first `10` invocation is finished and you incremented `tailCallId` and it is `10` but  `callIdIncrementer` is `11`. This is the inconsistency I have mentioned.